flaky test: TestClientSlowCacheRootfsRef/worker=containerd-snapshotter-stargz

https://api.travis-ci.org/v3/job/743987816/log.txt
https://api.travis-ci.org/v3/job/742351488/log.txt

```
--- FAIL: TestClientGatewayIntegration/TestClientSlowCacheRootfsRef/worker=containerd-snapshotter-stargz (0.53s)
        build_test.go:991: 
            	Error Trace:	build_test.go:991
            	            				client.go:188
            	            				build.go:56
            	            				solve.go:224
            	            				errgroup.go:57
            	            				asm_amd64.s:1357
            	Error:      	Received unexpected error:
            	            	rpc error: code = Canceled desc = context canceled
            	            	github.com/moby/buildkit/util/stack.Enable
            	            		/src/util/stack/stack.go:77
            	            	github.com/moby/buildkit/util/grpcerrors.FromGRPC
            	            		/src/util/grpcerrors/grpcerrors.go:188
            	            	github.com/moby/buildkit/util/grpcerrors.UnaryClientInterceptor
            	            		/src/util/grpcerrors/intercept.go:41
            	            	google.golang.org/grpc.(*ClientConn).Invoke
            	            		/src/vendor/google.golang.org/grpc/call.go:35
            	            	github.com/moby/buildkit/frontend/gateway/pb.(*lLBBridgeClient).StatFile
            	            		/src/frontend/gateway/pb/gateway.pb.go:2253
            	            	github.com/moby/buildkit/client.(*gatewayClientForBuild).StatFile
            	            		/src/client/build.go:107
            	            	github.com/moby/buildkit/frontend/gateway/grpcclient.(*reference).StatFile
            	            		/src/frontend/gateway/grpcclient/client.go:966
            	            	github.com/moby/buildkit/client.testClientSlowCacheRootfsRef.func1
            	            		/src/client/build_test.go:988
            	            	github.com/moby/buildkit/frontend/gateway/grpcclient.(*grpcClient).Run
            	            		/src/frontend/gateway/grpcclient/client.go:188
            	            	github.com/moby/buildkit/client.(*Client).Build.func2
            	            		/src/client/build.go:56
            	            	github.com/moby/buildkit/client.(*Client).solve.func3
            	            		/src/client/solve.go:224
            	            	golang.org/x/sync/errgroup.(*Group).Go.func1
            	            		/src/vendor/golang.org/x/sync/errgroup/errgroup.go:57
            	            	runtime.goexit
            	            		/usr/local/go/src/runtime/asm_amd64.s:1357
            	Test:       	TestClientGatewayIntegration/TestClientSlowCacheRootfsRef/worker=containerd-snapshotter-stargz
        sandbox.go:220: stdout: /usr/bin/containerd-stargz-grpc
        sandbox.go:220: stderr: /usr/bin/containerd-stargz-grpc
        sandbox.go:223: > startCmd 2020-11-09 04:54:25.577522578 +0000 UTC m=+27.442006107 [containerd-stargz-grpc --log-level debug --address /tmp/bktest_containerd_stargz_grpc507703150/containerd-stargz-grpc.sock --root /tmp/bktest_containerd_stargz_grpc507703150/root]
        sandbox.go:223: {"key":"buildkit/1/o5m0wc0jahobh7gdl0g8tkas6","level":"debug","msg":"checking layer availability","time":"2020-11-09T04:54:26.053806179Z"}
        sandbox.go:223: {"key":"buildkit/1/o5m0wc0jahobh7gdl0g8tkas6","level":"debug","mount-point":"/tmp/bktest_containerd_stargz_grpc507703150/root/snapshotter/snapshots/1/fs","msg":"layer is normal snapshot(overlayfs)","time":"2020-11-09T04:54:26.054006564Z"}
        sandbox.go:223: {"key":"buildkit/1/o5m0wc0jahobh7gdl0g8tkas6","level":"debug","msg":"checking layer availability","time":"2020-11-09T04:54:26.067420398Z"}
        sandbox.go:223: {"key":"buildkit/1/o5m0wc0jahobh7gdl0g8tkas6","level":"debug","mount-point":"/tmp/bktest_containerd_stargz_grpc507703150/root/snapshotter/snapshots/1/fs","msg":"layer is normal snapshot(overlayfs)","time":"2020-11-09T04:54:26.067519455Z"}
        sandbox.go:223: {"level":"debug","msg":"cleanup: dirs=[/tmp/bktest_containerd_stargz_grpc507703150/root/snapshotter/snapshots/1]","time":"2020-11-09T04:54:26.088063211Z"}
        sandbox.go:223: {"dir":"/tmp/bktest_containerd_stargz_grpc507703150/root/snapshotter/snapshots/1/fs","error":"specified path \"/tmp/bktest_containerd_stargz_grpc507703150/root/snapshotter/snapshots/1/fs\" isn't a mountpoint","level":"debug","msg":"failed to unmount","time":"2020-11-09T04:54:26.088101653Z"}
        sandbox.go:223: > sending sigterm 2020-11-09 04:54:26.101271697 +0000 UTC m=+27.965755229
        sandbox.go:223: > stopped 2020-11-09 04:54:26.102791573 +0000 UTC m=+27.967275209 signal: terminated -1
        sandbox.go:220: stdout: /usr/bin/containerd
        sandbox.go:220: stderr: /usr/bin/containerd
        sandbox.go:223: > startCmd 2020-11-09 04:54:25.628116126 +0000 UTC m=+27.492599659 [containerd --config /tmp/bktest_containerd425255307/config.toml]
        sandbox.go:223: time="2020-11-09T04:54:25.655553172Z" level=info msg="starting containerd" revision=09814d48d50816305a8e6c1a4ae3e2bcc4ba725a version=v1.4.0
        sandbox.go:223: time="2020-11-09T04:54:25.700507639Z" level=info msg="loading plugin \"io.containerd.content.v1.content\"..." type=io.containerd.content.v1
        sandbox.go:223: time="2020-11-09T04:54:25.700642759Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.aufs\"..." type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-09T04:54:25.700804975Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.btrfs\"..." type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-09T04:54:25.701121876Z" level=info msg="skip loading plugin \"io.containerd.snapshotter.v1.btrfs\"..." error="path /tmp/bktest_containerd425255307/root/io.containerd.snapshotter.v1.btrfs (ext4) must be a btrfs filesystem to be used with the btrfs snapshotter: skip plugin" type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-09T04:54:25.701184645Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.devmapper\"..." type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-09T04:54:25.701255017Z" level=warning msg="failed to load plugin io.containerd.snapshotter.v1.devmapper" error="devmapper not configured"
        sandbox.go:223: time="2020-11-09T04:54:25.701285460Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.native\"..." type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-09T04:54:25.701370780Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.overlayfs\"..." type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-09T04:54:25.701602905Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.zfs\"..." type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-09T04:54:25.702319278Z" level=info msg="skip loading plugin \"io.containerd.snapshotter.v1.zfs\"..." error="path /tmp/bktest_containerd425255307/root/io.containerd.snapshotter.v1.zfs must be a zfs filesystem to be used with the zfs snapshotter: skip plugin" type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-09T04:54:25.702496864Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.stargz\"..." type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-09T04:54:25.702614515Z" level=info msg="loading plugin \"io.containerd.metadata.v1.bolt\"..." type=io.containerd.metadata.v1
        sandbox.go:223: time="2020-11-09T04:54:25.704379499Z" level=warning msg="could not use snapshotter devmapper in metadata plugin" error="devmapper not configured"
        sandbox.go:223: time="2020-11-09T04:54:25.704404116Z" level=info msg="metadata content store policy set" policy=shared
        sandbox.go:223: time="2020-11-09T04:54:25.709562164Z" level=info msg="loading plugin \"io.containerd.differ.v1.walking\"..." type=io.containerd.differ.v1
        sandbox.go:223: time="2020-11-09T04:54:25.709593762Z" level=info msg="loading plugin \"io.containerd.gc.v1.scheduler\"..." type=io.containerd.gc.v1
        sandbox.go:223: time="2020-11-09T04:54:25.709642010Z" level=info msg="loading plugin \"io.containerd.service.v1.introspection-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-09T04:54:25.709698226Z" level=info msg="loading plugin \"io.containerd.service.v1.containers-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-09T04:54:25.709719556Z" level=info msg="loading plugin \"io.containerd.service.v1.content-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-09T04:54:25.709758253Z" level=info msg="loading plugin \"io.containerd.service.v1.diff-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-09T04:54:25.709965516Z" level=info msg="loading plugin \"io.containerd.service.v1.images-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710008421Z" level=info msg="loading plugin \"io.containerd.service.v1.leases-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710034588Z" level=info msg="loading plugin \"io.containerd.service.v1.namespaces-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710057257Z" level=info msg="loading plugin \"io.containerd.service.v1.snapshots-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710081504Z" level=info msg="loading plugin \"io.containerd.runtime.v1.linux\"..." type=io.containerd.runtime.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710203247Z" level=info msg="loading plugin \"io.containerd.runtime.v2.task\"..." type=io.containerd.runtime.v2
        sandbox.go:223: time="2020-11-09T04:54:25.710300225Z" level=info msg="loading plugin \"io.containerd.monitor.v1.cgroups\"..." type=io.containerd.monitor.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710711403Z" level=info msg="loading plugin \"io.containerd.service.v1.tasks-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710746823Z" level=info msg="loading plugin \"io.containerd.internal.v1.restart\"..." type=io.containerd.internal.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710805126Z" level=info msg="loading plugin \"io.containerd.grpc.v1.containers\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710830560Z" level=info msg="loading plugin \"io.containerd.grpc.v1.content\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710858416Z" level=info msg="loading plugin \"io.containerd.grpc.v1.diff\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710871451Z" level=info msg="loading plugin \"io.containerd.grpc.v1.events\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710885357Z" level=info msg="loading plugin \"io.containerd.grpc.v1.healthcheck\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710934592Z" level=info msg="loading plugin \"io.containerd.grpc.v1.images\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710966897Z" level=info msg="loading plugin \"io.containerd.grpc.v1.leases\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710980142Z" level=info msg="loading plugin \"io.containerd.grpc.v1.namespaces\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-09T04:54:25.710992173Z" level=info msg="loading plugin \"io.containerd.internal.v1.opt\"..." type=io.containerd.internal.v1
        sandbox.go:223: time="2020-11-09T04:54:25.711752810Z" level=info msg="loading plugin \"io.containerd.grpc.v1.snapshots\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-09T04:54:25.711775337Z" level=info msg="loading plugin \"io.containerd.grpc.v1.tasks\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-09T04:54:25.711791180Z" level=info msg="loading plugin \"io.containerd.grpc.v1.version\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-09T04:54:25.711804989Z" level=info msg="loading plugin \"io.containerd.grpc.v1.introspection\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-09T04:54:25.711997158Z" level=info msg=serving... address=/tmp/bktest_containerd425255307/debug.sock
        sandbox.go:223: time="2020-11-09T04:54:25.712046292Z" level=info msg=serving... address=/tmp/bktest_containerd425255307/containerd.sock.ttrpc
        sandbox.go:223: time="2020-11-09T04:54:25.712081912Z" level=info msg=serving... address=/tmp/bktest_containerd425255307/containerd.sock
        sandbox.go:223: time="2020-11-09T04:54:25.712094174Z" level=debug msg="sd notification" error="<nil>" notified=false state="READY=1"
        sandbox.go:223: time="2020-11-09T04:54:25.712105755Z" level=info msg="containerd successfully booted in 0.057173s"
        sandbox.go:223: time="2020-11-09T04:54:25.732232230Z" level=debug msg="received signal" signal="broken pipe"
        sandbox.go:223: time="2020-11-09T04:54:25.811143753Z" level=debug msg="garbage collected" d=1.242468ms
        sandbox.go:223: time="2020-11-09T04:54:26.031569338Z" level=debug msg="(*service).Write started" expected="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" ref="manifest-sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" total=527
        sandbox.go:223: time="2020-11-09T04:54:26.034729457Z" level=debug msg="prepare snapshot" key=o5m0wc0jahobh7gdl0g8tkas6 parent=
        sandbox.go:223: time="2020-11-09T04:54:26.048880311Z" level=debug msg="event published" ns=buildkit topic=/snapshot/prepare type=containerd.events.SnapshotPrepare
        sandbox.go:223: time="2020-11-09T04:54:26.050465056Z" level=debug msg="(*service).Write started" expected="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f" ref="config-sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f" total=1493
        sandbox.go:223: time="2020-11-09T04:54:26.053455578Z" level=debug msg="get snapshot mounts" key=o5m0wc0jahobh7gdl0g8tkas6
        sandbox.go:223: time="2020-11-09T04:54:26.064569480Z" level=debug msg="get snapshot mounts" key=o5m0wc0jahobh7gdl0g8tkas6
        sandbox.go:223: time="2020-11-09T04:54:26.083720802Z" level=debug msg="remove snapshot" key=o5m0wc0jahobh7gdl0g8tkas6 snapshotter=stargz
        sandbox.go:223: time="2020-11-09T04:54:26.083797651Z" level=debug msg="remove content" key="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc"
        sandbox.go:223: time="2020-11-09T04:54:26.083826038Z" level=debug msg="remove content" key="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f"
        sandbox.go:223: time="2020-11-09T04:54:26.084742517Z" level=debug msg="schedule snapshotter cleanup" snapshotter=stargz
        sandbox.go:223: time="2020-11-09T04:54:26.084769320Z" level=debug msg="schedule content cleanup"
        sandbox.go:223: time="2020-11-09T04:54:26.084930116Z" level=debug msg="removed content" digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc"
        sandbox.go:223: time="2020-11-09T04:54:26.085012851Z" level=debug msg="removed content" digest="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f"
        sandbox.go:223: time="2020-11-09T04:54:26.085054294Z" level=debug msg="content garbage collected" d="266Âµs"
        sandbox.go:223: time="2020-11-09T04:54:26.087836861Z" level=debug msg="removed snapshot" key=buildkit/1/o5m0wc0jahobh7gdl0g8tkas6 snapshotter=stargz
        sandbox.go:223: time="2020-11-09T04:54:26.088556221Z" level=debug msg="snapshot garbage collected" d=3.47376ms snapshotter=stargz
        sandbox.go:223: time="2020-11-09T04:54:26.088596532Z" level=debug msg="garbage collected" d=1.111866ms
        sandbox.go:223: > sending sigterm 2020-11-09 04:54:26.098768407 +0000 UTC m=+27.963251923
        sandbox.go:223: time="2020-11-09T04:54:26.098844368Z" level=debug msg="received signal" signal=terminated
        sandbox.go:223: time="2020-11-09T04:54:26.098867603Z" level=debug msg="sd notification" error="<nil>" notified=false state="STOPPING=1"
        sandbox.go:223: > stopped 2020-11-09 04:54:26.101244942 +0000 UTC m=+27.965728472 exit status 0 0
        sandbox.go:220: stdout: /usr/bin/buildkitd
        sandbox.go:220: stderr: /usr/bin/buildkitd
        sandbox.go:223: > startCmd 2020-11-09 04:54:25.732103126 +0000 UTC m=+27.596586663 [buildkitd --oci-worker=false --containerd-worker-gc=false --containerd-worker=true --containerd-worker-addr /tmp/bktest_containerd425255307/containerd.sock --containerd-worker-labels=org.mobyproject.buildkit.worker.sandbox=true --containerd-worker-snapshotter=stargz --config=/tmp/bktest_config528370652/buildkitd.toml --root /tmp/bktest_buildkitd453096949 --addr unix:///tmp/bktest_buildkitd453096949/buildkitd.sock --debug]
        sandbox.go:223: time="2020-11-09T04:54:25Z" level=info msg="found worker \"kijeq1mhs8frvu7zmp7csatid\", labels=map[org.mobyproject.buildkit.worker.executor:containerd org.mobyproject.buildkit.worker.hostname:0e77e4bc466f org.mobyproject.buildkit.worker.sandbox:true org.mobyproject.buildkit.worker.snapshotter:stargz], platforms=[linux/amd64 linux/arm64 linux/riscv64 linux/ppc64le linux/s390x linux/386 linux/arm/v7 linux/arm/v6]"
        sandbox.go:223: time="2020-11-09T04:54:25Z" level=info msg="found 1 workers, default=\"kijeq1mhs8frvu7zmp7csatid\""
        sandbox.go:223: time="2020-11-09T04:54:25Z" level=warning msg="currently, only the default worker can be used."
        sandbox.go:223: time="2020-11-09T04:54:25Z" level=info msg="running server on /tmp/bktest_buildkitd453096949/buildkitd.sock"
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=debug msg="session started"
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=debug msg=resolving host="localhost:46813"
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=debug msg="do request" host="localhost:46813" request.header.accept="application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, */*" request.header.user-agent=containerd/1.4.0+unknown request.method=HEAD url="http://localhost:46813/v2/library/busybox/manifests/latest?ns=docker.io"
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=debug msg="fetch response received" host="localhost:46813" response.header.content-length=527 response.header.content-type=application/vnd.docker.distribution.manifest.v2+json response.header.date="Mon, 09 Nov 2020 04:54:26 GMT" response.header.docker-content-digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" response.header.docker-distribution-api-version=registry/2.0 response.header.etag="\"sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc\"" response.status="200 OK" url="http://localhost:46813/v2/library/busybox/manifests/latest?ns=docker.io"
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=debug msg=resolved desc.digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" host="localhost:46813"
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=debug msg=fetch digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" mediatype=application/vnd.docker.distribution.manifest.v2+json size=527
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=debug msg="do request" digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" mediatype=application/vnd.docker.distribution.manifest.v2+json request.header.accept="application/vnd.docker.distribution.manifest.v2+json, */*" request.header.user-agent=containerd/1.4.0+unknown request.method=GET size=527 url="http://localhost:46813/v2/library/busybox/manifests/sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc?ns=docker.io"
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=debug msg="fetch response received" digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" mediatype=application/vnd.docker.distribution.manifest.v2+json response.header.content-length=527 response.header.content-type=application/vnd.docker.distribution.manifest.v2+json response.header.date="Mon, 09 Nov 2020 04:54:26 GMT" response.header.docker-content-digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" response.header.docker-distribution-api-version=registry/2.0 response.header.etag="\"sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc\"" response.status="200 OK" size=527 url="http://localhost:46813/v2/library/busybox/manifests/sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc?ns=docker.io"
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=debug msg=fetch digest="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f" mediatype=application/vnd.docker.container.image.v1+json size=1493
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=debug msg="do request" digest="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f" mediatype=application/vnd.docker.container.image.v1+json request.header.accept="application/vnd.docker.container.image.v1+json, */*" request.header.user-agent=containerd/1.4.0+unknown request.method=GET size=1493 url="http://localhost:46813/v2/library/busybox/blobs/sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f?ns=docker.io"
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=debug msg="fetch response received" digest="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f" mediatype=application/vnd.docker.container.image.v1+json response.header.accept-ranges=bytes response.header.cache-control="max-age=31536000" response.header.content-length=1493 response.header.content-type=application/octet-stream response.header.date="Mon, 09 Nov 2020 04:54:26 GMT" response.header.docker-content-digest="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f" response.header.docker-distribution-api-version=registry/2.0 response.header.etag="\"sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f\"" response.status="200 OK" size=1493 url="http://localhost:46813/v2/library/busybox/blobs/sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f?ns=docker.io"
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=error msg="/moby.buildkit.v1.frontend.LLBBridge/StatFile returned error: rpc error: code = Unknown desc = failed to compute cache key: \"/notfound\" not found: not found\n"
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=error msg="/moby.buildkit.v1.frontend.LLBBridge/StatFile returned error: rpc error: code = Canceled desc = context canceled\n"
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=debug msg="session finished: <nil>"
        sandbox.go:223: > sending sigterm 2020-11-09 04:54:26.094215204 +0000 UTC m=+27.958698725
        sandbox.go:223: time="2020-11-09T04:54:26Z" level=info msg="stopping server"
        sandbox.go:223: buildkitd: context canceled
        sandbox.go:223: > stopped 2020-11-09 04:54:26.098417125 +0000 UTC m=+27.962900659 exit status 1 1
```

@ktock @hinshun

Not only stargz

```
    --- FAIL: TestClientGatewayIntegration/TestClientSlowCacheRootfsRef/worker=containerd-1.3 (0.59s)
        build_test.go:995: 
            	Error Trace:	build_test.go:995
            	            				client.go:187
            	            				build.go:56
            	            				solve.go:224
            	            				errgroup.go:57
            	            				asm_amd64.s:1357
            	Error:      	Received unexpected error:
            	            	rpc error: code = Canceled desc = context canceled
            	            	github.com/moby/buildkit/util/stack.Enable
            	            		/src/util/stack/stack.go:77
            	            	github.com/moby/buildkit/util/grpcerrors.FromGRPC
            	            		/src/util/grpcerrors/grpcerrors.go:188
            	            	github.com/moby/buildkit/util/grpcerrors.UnaryClientInterceptor
            	            		/src/util/grpcerrors/intercept.go:41
            	            	google.golang.org/grpc.(*ClientConn).Invoke
            	            		/src/vendor/google.golang.org/grpc/call.go:35
            	            	github.com/moby/buildkit/frontend/gateway/pb.(*lLBBridgeClient).StatFile
            	            		/src/frontend/gateway/pb/gateway.pb.go:2262
            	            	github.com/moby/buildkit/client.(*gatewayClientForBuild).StatFile
            	            		/src/client/build.go:107
            	            	github.com/moby/buildkit/frontend/gateway/grpcclient.(*reference).StatFile
            	            		/src/frontend/gateway/grpcclient/client.go:997
            	            	github.com/moby/buildkit/client.testClientSlowCacheRootfsRef.func1
            	            		/src/client/build_test.go:992
            	            	github.com/moby/buildkit/frontend/gateway/grpcclient.(*grpcClient).Run
            	            		/src/frontend/gateway/grpcclient/client.go:187
            	            	github.com/moby/buildkit/client.(*Client).Build.func2
            	            		/src/client/build.go:56
            	            	github.com/moby/buildkit/client.(*Client).solve.func3
            	            		/src/client/solve.go:224
            	            	golang.org/x/sync/errgroup.(*Group).Go.func1
            	            		/src/vendor/golang.org/x/sync/errgroup/errgroup.go:57
            	            	runtime.goexit
            	            		/usr/local/go/src/runtime/asm_amd64.s:1357
            	Test:       	TestClientGatewayIntegration/TestClientSlowCacheRootfsRef/worker=containerd-1.3
        sandbox.go:220: stdout: /opt/containerd-alt/bin/containerd
        sandbox.go:220: stderr: /opt/containerd-alt/bin/containerd
        sandbox.go:223: > startCmd 2020-11-21 10:32:48.624944631 +0000 UTC m=+30.366284246 [/opt/containerd-alt/bin/containerd --config /tmp/bktest_containerd036298913/config.toml]
        sandbox.go:223: time="2020-11-21T10:32:48.657884045Z" level=info msg="starting containerd" revision=8fba4e9a7d01810a393d5d25a3621dc101981175 version=v1.3.7
        sandbox.go:223: time="2020-11-21T10:32:48.681263270Z" level=info msg="loading plugin \"io.containerd.content.v1.content\"..." type=io.containerd.content.v1
        sandbox.go:223: time="2020-11-21T10:32:48.681394235Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.btrfs\"..." type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-21T10:32:48.681668854Z" level=info msg="skip loading plugin \"io.containerd.snapshotter.v1.btrfs\"..." error="path /tmp/bktest_containerd036298913/root/io.containerd.snapshotter.v1.btrfs (ext4) must be a btrfs filesystem to be used with the btrfs snapshotter: skip plugin" type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-21T10:32:48.681699766Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.devmapper\"..." type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-21T10:32:48.681721028Z" level=warning msg="failed to load plugin io.containerd.snapshotter.v1.devmapper" error="devmapper not configured"
        sandbox.go:223: time="2020-11-21T10:32:48.681738396Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.aufs\"..." type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-21T10:32:48.681798975Z" level=info msg="skip loading plugin \"io.containerd.snapshotter.v1.aufs\"..." error="modprobe aufs failed: \"\": exec: \"modprobe\": executable file not found in $PATH: skip plugin" type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-21T10:32:48.681830863Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.native\"..." type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-21T10:32:48.681904112Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.overlayfs\"..." type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-21T10:32:48.682041296Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.zfs\"..." type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-21T10:32:48.682260688Z" level=info msg="skip loading plugin \"io.containerd.snapshotter.v1.zfs\"..." error="path /tmp/bktest_containerd036298913/root/io.containerd.snapshotter.v1.zfs must be a zfs filesystem to be used with the zfs snapshotter: skip plugin" type=io.containerd.snapshotter.v1
        sandbox.go:223: time="2020-11-21T10:32:48.682284253Z" level=info msg="loading plugin \"io.containerd.metadata.v1.bolt\"..." type=io.containerd.metadata.v1
        sandbox.go:223: time="2020-11-21T10:32:48.682332177Z" level=warning msg="could not use snapshotter devmapper in metadata plugin" error="devmapper not configured"
        sandbox.go:223: time="2020-11-21T10:32:48.682349119Z" level=info msg="metadata content store policy set" policy=shared
        sandbox.go:223: time="2020-11-21T10:32:48.688365723Z" level=info msg="loading plugin \"io.containerd.differ.v1.walking\"..." type=io.containerd.differ.v1
        sandbox.go:223: time="2020-11-21T10:32:48.688398718Z" level=info msg="loading plugin \"io.containerd.gc.v1.scheduler\"..." type=io.containerd.gc.v1
        sandbox.go:223: time="2020-11-21T10:32:48.688457952Z" level=info msg="loading plugin \"io.containerd.service.v1.containers-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-21T10:32:48.688480682Z" level=info msg="loading plugin \"io.containerd.service.v1.content-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-21T10:32:48.688500452Z" level=info msg="loading plugin \"io.containerd.service.v1.diff-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-21T10:32:48.688518217Z" level=info msg="loading plugin \"io.containerd.service.v1.images-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-21T10:32:48.688533310Z" level=info msg="loading plugin \"io.containerd.service.v1.leases-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-21T10:32:48.688555038Z" level=info msg="loading plugin \"io.containerd.service.v1.namespaces-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-21T10:32:48.688576450Z" level=info msg="loading plugin \"io.containerd.service.v1.snapshots-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-21T10:32:48.688596590Z" level=info msg="loading plugin \"io.containerd.runtime.v1.linux\"..." type=io.containerd.runtime.v1
        sandbox.go:223: time="2020-11-21T10:32:48.688713403Z" level=info msg="loading plugin \"io.containerd.runtime.v2.task\"..." type=io.containerd.runtime.v2
        sandbox.go:223: time="2020-11-21T10:32:48.688803251Z" level=info msg="loading plugin \"io.containerd.monitor.v1.cgroups\"..." type=io.containerd.monitor.v1
        sandbox.go:223: time="2020-11-21T10:32:48.689814650Z" level=info msg="loading plugin \"io.containerd.service.v1.tasks-service\"..." type=io.containerd.service.v1
        sandbox.go:223: time="2020-11-21T10:32:48.689869798Z" level=info msg="loading plugin \"io.containerd.internal.v1.restart\"..." type=io.containerd.internal.v1
        sandbox.go:223: time="2020-11-21T10:32:48.689933637Z" level=info msg="loading plugin \"io.containerd.grpc.v1.containers\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-21T10:32:48.689960119Z" level=info msg="loading plugin \"io.containerd.grpc.v1.content\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-21T10:32:48.689986162Z" level=info msg="loading plugin \"io.containerd.grpc.v1.diff\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-21T10:32:48.690008263Z" level=info msg="loading plugin \"io.containerd.grpc.v1.events\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-21T10:32:48.690037061Z" level=info msg="loading plugin \"io.containerd.grpc.v1.healthcheck\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-21T10:32:48.690055549Z" level=info msg="loading plugin \"io.containerd.grpc.v1.images\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-21T10:32:48.690077151Z" level=info msg="loading plugin \"io.containerd.grpc.v1.leases\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-21T10:32:48.690099910Z" level=info msg="loading plugin \"io.containerd.grpc.v1.namespaces\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-21T10:32:48.690124733Z" level=info msg="loading plugin \"io.containerd.internal.v1.opt\"..." type=io.containerd.internal.v1
        sandbox.go:223: time="2020-11-21T10:32:48.690179766Z" level=info msg="loading plugin \"io.containerd.grpc.v1.snapshots\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-21T10:32:48.690204653Z" level=info msg="loading plugin \"io.containerd.grpc.v1.tasks\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-21T10:32:48.690228279Z" level=info msg="loading plugin \"io.containerd.grpc.v1.version\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-21T10:32:48.690252079Z" level=info msg="loading plugin \"io.containerd.grpc.v1.introspection\"..." type=io.containerd.grpc.v1
        sandbox.go:223: time="2020-11-21T10:32:48.690481588Z" level=info msg=serving... address=/tmp/bktest_containerd036298913/debug.sock
        sandbox.go:223: time="2020-11-21T10:32:48.690534099Z" level=info msg=serving... address=/tmp/bktest_containerd036298913/containerd.sock.ttrpc
        sandbox.go:223: time="2020-11-21T10:32:48.690568595Z" level=info msg=serving... address=/tmp/bktest_containerd036298913/containerd.sock
        sandbox.go:223: time="2020-11-21T10:32:48.690587046Z" level=info msg="containerd successfully booted in 0.033272s"
        sandbox.go:223: time="2020-11-21T10:32:48.726458912Z" level=debug msg="received signal" signal="broken pipe"
        sandbox.go:223: time="2020-11-21T10:32:48.789970590Z" level=debug msg="garbage collected" d=1.397744ms
        sandbox.go:223: time="2020-11-21T10:32:49.134695387Z" level=debug msg="(*service).Write started" expected="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" ref="manifest-sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" total=527
        sandbox.go:223: time="2020-11-21T10:32:49.137498599Z" level=debug msg="prepare snapshot" key=zdcwhjec7l7i1iatjm08uo6j7 parent=
        sandbox.go:223: time="2020-11-21T10:32:49.152720984Z" level=debug msg="event published" ns=buildkit topic=/snapshot/prepare type=containerd.events.SnapshotPrepare
        sandbox.go:223: time="2020-11-21T10:32:49.161015681Z" level=debug msg="(*service).Write started" expected="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f" ref="config-sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f" total=1493
        sandbox.go:223: time="2020-11-21T10:32:49.161750916Z" level=debug msg="get snapshot mounts" key=zdcwhjec7l7i1iatjm08uo6j7
        sandbox.go:223: time="2020-11-21T10:32:49.171916956Z" level=debug msg="get snapshot mounts" key=zdcwhjec7l7i1iatjm08uo6j7
        sandbox.go:223: time="2020-11-21T10:32:49.201876182Z" level=debug msg="remove snapshot" key=zdcwhjec7l7i1iatjm08uo6j7 snapshotter=overlayfs
        sandbox.go:223: time="2020-11-21T10:32:49.201956480Z" level=debug msg="remove content" key="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc"
        sandbox.go:223: time="2020-11-21T10:32:49.201977421Z" level=debug msg="remove content" key="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f"
        sandbox.go:223: time="2020-11-21T10:32:49.202995872Z" level=debug msg="schedule snapshotter cleanup" snapshotter=overlayfs
        sandbox.go:223: time="2020-11-21T10:32:49.203018113Z" level=debug msg="schedule content cleanup"
        sandbox.go:223: time="2020-11-21T10:32:49.203182968Z" level=debug msg="removed content" digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc"
        sandbox.go:223: time="2020-11-21T10:32:49.203252754Z" level=debug msg="removed content" digest="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f"
        sandbox.go:223: time="2020-11-21T10:32:49.203279257Z" level=debug msg="content garbage collected" d="232.525Âµs"
        sandbox.go:223: time="2020-11-21T10:32:49.205637158Z" level=debug msg="removed snapshot" key=buildkit/1/zdcwhjec7l7i1iatjm08uo6j7 snapshotter=overlayfs
        sandbox.go:223: time="2020-11-21T10:32:49.205857814Z" level=debug msg="snapshot garbage collected" d=2.561753ms snapshotter=overlayfs
        sandbox.go:223: time="2020-11-21T10:32:49.205885689Z" level=debug msg="garbage collected" d=1.205962ms
        sandbox.go:223: time="2020-11-21T10:32:49.206817618Z" level=debug msg="garbage collected" d="851.185Âµs"
        sandbox.go:223: > sending sigterm 2020-11-21 10:32:49.209953655 +0000 UTC m=+30.951293275
        sandbox.go:223: time="2020-11-21T10:32:49.210024762Z" level=debug msg="received signal" signal=terminated
        sandbox.go:223: > stopped 2020-11-21 10:32:49.211780788 +0000 UTC m=+30.953120413 exit status 0 0
        sandbox.go:220: stdout: /usr/bin/buildkitd
        sandbox.go:220: stderr: /usr/bin/buildkitd
        sandbox.go:223: > startCmd 2020-11-21 10:32:48.726325436 +0000 UTC m=+30.467665056 [buildkitd --oci-worker=false --containerd-worker-gc=false --containerd-worker=true --containerd-worker-addr /tmp/bktest_containerd036298913/containerd.sock --containerd-worker-labels=org.mobyproject.buildkit.worker.sandbox=true --config=/tmp/bktest_config150056906/buildkitd.toml --root /tmp/bktest_buildkitd157112716 --addr unix:///tmp/bktest_buildkitd157112716/buildkitd.sock --debug]
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=info msg="found worker \"m7pvo8ql7meuhiepr2lizfwe0\", labels=map[org.mobyproject.buildkit.worker.executor:containerd org.mobyproject.buildkit.worker.hostname:45a0975e52af org.mobyproject.buildkit.worker.sandbox:true org.mobyproject.buildkit.worker.snapshotter:overlayfs], platforms=[linux/amd64 linux/arm64 linux/riscv64 linux/ppc64le linux/s390x linux/386 linux/arm/v7 linux/arm/v6]"
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=info msg="found 1 workers, default=\"m7pvo8ql7meuhiepr2lizfwe0\""
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=warning msg="currently, only the default worker can be used."
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=info msg="running server on /tmp/bktest_buildkitd157112716/buildkitd.sock"
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=debug msg="session started"
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=debug msg=resolving host="localhost:46853"
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=debug msg="do request" host="localhost:46853" request.header.accept="application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, */*" request.header.user-agent=containerd/1.4.0+unknown request.method=HEAD url="http://localhost:46853/v2/library/busybox/manifests/latest?ns=docker.io"
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=debug msg="fetch response received" host="localhost:46853" response.header.content-length=527 response.header.content-type=application/vnd.docker.distribution.manifest.v2+json response.header.date="Sat, 21 Nov 2020 10:32:49 GMT" response.header.docker-content-digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" response.header.docker-distribution-api-version=registry/2.0 response.header.etag="\"sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc\"" response.status="200 OK" url="http://localhost:46853/v2/library/busybox/manifests/latest?ns=docker.io"
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=debug msg=resolved desc.digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" host="localhost:46853"
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=debug msg=fetch digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" mediatype=application/vnd.docker.distribution.manifest.v2+json size=527
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=debug msg="do request" digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" mediatype=application/vnd.docker.distribution.manifest.v2+json request.header.accept="application/vnd.docker.distribution.manifest.v2+json, */*" request.header.user-agent=containerd/1.4.0+unknown request.method=GET size=527 url="http://localhost:46853/v2/library/busybox/manifests/sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc?ns=docker.io"
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=debug msg="fetch response received" digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" mediatype=application/vnd.docker.distribution.manifest.v2+json response.header.content-length=527 response.header.content-type=application/vnd.docker.distribution.manifest.v2+json response.header.date="Sat, 21 Nov 2020 10:32:49 GMT" response.header.docker-content-digest="sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc" response.header.docker-distribution-api-version=registry/2.0 response.header.etag="\"sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc\"" response.status="200 OK" size=527 url="http://localhost:46853/v2/library/busybox/manifests/sha256:c9249fdf56138f0d929e2080ae98ee9cb2946f71498fc1484288e6a935b5e5bc?ns=docker.io"
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=debug msg=fetch digest="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f" mediatype=application/vnd.docker.container.image.v1+json size=1493
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=debug msg="do request" digest="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f" mediatype=application/vnd.docker.container.image.v1+json request.header.accept="application/vnd.docker.container.image.v1+json, */*" request.header.user-agent=containerd/1.4.0+unknown request.method=GET size=1493 url="http://localhost:46853/v2/library/busybox/blobs/sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f?ns=docker.io"
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=debug msg="fetch response received" digest="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f" mediatype=application/vnd.docker.container.image.v1+json response.header.accept-ranges=bytes response.header.cache-control="max-age=31536000" response.header.content-length=1493 response.header.content-type=application/octet-stream response.header.date="Sat, 21 Nov 2020 10:32:49 GMT" response.header.docker-content-digest="sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f" response.header.docker-distribution-api-version=registry/2.0 response.header.etag="\"sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f\"" response.status="200 OK" size=1493 url="http://localhost:46853/v2/library/busybox/blobs/sha256:f0b02e9d092d905d0d87a8455a1ae3e9bb47b4aa3dc125125ca5cd10d6441c9f?ns=docker.io"
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=error msg="/moby.buildkit.v1.frontend.LLBBridge/StatFile returned error: rpc error: code = Unknown desc = failed to compute cache key: \"/notfound\" not found: not found\n"
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=error msg="/moby.buildkit.v1.frontend.LLBBridge/StatFile returned error: rpc error: code = Canceled desc = context canceled\n"
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=debug msg="session finished: <nil>"
        sandbox.go:223: > sending sigterm 2020-11-21 10:32:49.20856232 +0000 UTC m=+30.949901945
        sandbox.go:223: time="2020-11-21T10:32:49Z" level=info msg="stopping server"
        sandbox.go:223: buildkitd: context canceled
        sandbox.go:223: > stopped 2020-11-21 10:32:49.209628232 +0000 UTC m=+30.950967851 exit status 1 1
```

--- a/client/build_test.go
+++ b/client/build_test.go
@@ -941,11 +941,6 @@ func testClientSlowCacheRootfsRef(t *testing.T, sb integration.Sandbox) {
 
 	ctx := context.TODO()
 
-	// https://github.com/moby/buildkit/issues/1820
-	if sb.ContainerdAddress() != "" {
-		t.Skip()
-	}
-
 	c, err := New(ctx, sb.Address())
 	require.NoError(t, err)
 	defer c.Close()

--- a/frontend/gateway/gateway.go
+++ b/frontend/gateway/gateway.go
@@ -255,7 +255,7 @@ func (gf *gatewayFrontend) Solve(ctx context.Context, llbBridge frontend.Fronten
 	err = w.Executor().Run(ctx, "", mountWithSession(rootFS, session.NewGroup(sid)), nil, executor.ProcessInfo{Meta: meta, Stdin: lbf.Stdin, Stdout: lbf.Stdout, Stderr: os.Stderr}, nil)
 
 	if err != nil {
-		if errors.Is(err, context.Canceled) && lbf.isErrServerClosed {
+		if errdefs.IsCanceled(err) && lbf.isErrServerClosed {
 			err = errors.Errorf("frontend grpc server closed unexpectedly")
 		}
 		// An existing error (set via Return rpc) takes

--- a/solver/errdefs/context.go
+++ b/solver/errdefs/context.go
@@ -1,0 +1,13 @@
+package errdefs
+
+import (
+	"context"
+	"errors"
+
+	"github.com/moby/buildkit/util/grpcerrors"
+	"google.golang.org/grpc/codes"
+)
+
+func IsCanceled(err error) bool {
+	return errors.Is(err, context.Canceled) || grpcerrors.Code(err) == codes.Canceled
+}

--- a/source/containerimage/pull.go
+++ b/source/containerimage/pull.go
@@ -10,18 +10,19 @@ import (
 
 	"github.com/containerd/containerd/content"
 	"github.com/containerd/containerd/diff"
+	containerderrdefs "github.com/containerd/containerd/errdefs"
 	"github.com/containerd/containerd/images"
 	ctdlabels "github.com/containerd/containerd/labels"
 	"github.com/containerd/containerd/leases"
 	"github.com/containerd/containerd/platforms"
 	"github.com/containerd/containerd/remotes/docker"
 	"github.com/containerd/containerd/snapshots"
-	"github.com/docker/docker/errdefs"
 	"github.com/moby/buildkit/cache"
 	"github.com/moby/buildkit/client/llb"
 	"github.com/moby/buildkit/session"
 	"github.com/moby/buildkit/snapshot"
 	"github.com/moby/buildkit/solver"
+	"github.com/moby/buildkit/solver/errdefs"
 	"github.com/moby/buildkit/source"
 	"github.com/moby/buildkit/util/flightcontrol"
 	"github.com/moby/buildkit/util/imageutil"
@@ -177,7 +178,7 @@ func (p *puller) CacheKey(ctx context.Context, g session.Group, index int) (cach
 			return nil, p.cacheKeyErr
 		}
 		defer func() {
-			if !errors.Is(err, context.Canceled) {
+			if !errdefs.IsCanceled(err) {
 				p.cacheKeyErr = err
 			}
 		}()
@@ -186,12 +187,7 @@ func (p *puller) CacheKey(ctx context.Context, g session.Group, index int) (cach
 			return nil, err
 		}
 		p.releaseTmpLeases = done
-		imageutil.AddLease(p.releaseTmpLeases)
-		defer func() {
-			if err != nil {
-				p.releaseTmpLeases(ctx)
-			}
-		}()
+		defer imageutil.AddLease(done)
 
 		resolveProgressDone := oneOffProgress(ctx, "resolve "+p.Src.String())
 		defer func() {
@@ -283,7 +279,11 @@ func (p *puller) Snapshot(ctx context.Context, g session.Group) (ir cache.Immuta
 	if len(p.manifest.Descriptors) == 0 {
 		return nil, nil
 	}
-	defer p.releaseTmpLeases(ctx)
+	defer func() {
+		if p.releaseTmpLeases != nil {
+			p.releaseTmpLeases(context.TODO())
+		}
+	}()
 
 	var current cache.ImmutableRef
 	defer func() {
@@ -306,7 +306,7 @@ func (p *puller) Snapshot(ctx context.Context, g session.Group) (ir cache.Immuta
 	}
 
 	for _, desc := range p.manifest.Nonlayers {
-		if _, err := p.ContentStore.Info(ctx, desc.Digest); errdefs.IsNotFound(err) {
+		if _, err := p.ContentStore.Info(ctx, desc.Digest); containerderrdefs.IsNotFound(err) {
 			// manifest or config must have gotten gc'd after CacheKey, re-pull them
 			ctx, done, err := leaseutil.WithLease(ctx, p.LeaseManager, leaseutil.MakeTemporary)
 			if err != nil {

